### PR TITLE
Integration test MPI_VISCOUS_TAYLOR_GREEN_VORTEX_ENERGY_CHECK

### PR DESCRIPTION
**Pull request by jbrillon**
Pull request to incorporate the addition of a integration test for viscous Taylor Green Vortex (TGV) which computes the kinetic energy at some final time, using the `flow_solver` to setup and advance the solution, and compares the resulting value with an expected value, checking that the relative error is within some hard-coded tolerance.

### CODE REVIEW

File: src/testing/taylor_green_vortex_energy_check.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    std::unique_ptr<FlowSolverCaseBase<dim, nstate>> periodic_cube_flow = std::make_shared<PeriodicCubeFlow<dim,nstate>>(parameters_input);
+
+    // Integrate to final time
+    flow_solver->run_test();
+
+    // Compute kinetic energy
+    const double kinetic_energy_computed = periodic_cube_flow->integrate_over_domain(*(flow_solver->dg),"kinetic_energy");
+    
+    // Hard-coded expected energy value
+    const double kinetic_energy_expected = 1.2074066465958100e-01;
```
Comment (dougshidong) [inline src/testing/taylor_green_vortex_energy_check.cpp:25]: Can this expected energy value be evaluated based on final time? Or is this just based off some literature? If so, please cite the source.

File: tests/integration_tests_control_files/flow_solver/viscous_taylor_green_vortex_energy_check.prm [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    set runge_kutta_order = 3
+  end
+end
+
+# freestream Mach number
+subsection euler
+  set mach_infinity = 0.1
+end
+
+# freestream Reynolds number and Prandtl number
```
Comment (dougshidong) [inline tests/integration_tests_control_files/flow_solver/viscous_taylor_green_vortex_energy_check.prm:35]: Add reference where you got those numbers.

**Review by dougshidong — APPROVED** [review]

File: src/testing/taylor_green_vortex_energy_check.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// #include <deal.II/grid/grid_generator.h>
+// #include <deal.II/grid/grid_refinement.h>
+// #include <deal.II/grid/grid_tools.h>
+// #include <deal.II/grid/grid_out.h>
+// #include <deal.II/grid/grid_in.h>
+// #include "parameters/all_parameters.h"
+// #include "dg/dg_factory.hpp"
+// #include "ode_solver/ode_solver_factory.h"
+// #include "flow_solver.h"
+// #include <fstream>
```
[the remark this answers is not in the document]
Comment (jbrillon) [reply src/testing/taylor_green_vortex_energy_check.h:20]: sorry this is draft, i was planning on cleaning it all up before marking it as open

File: src/testing/taylor_green_vortex_energy_check.cpp (hunk 1 above)
Comment (jbrillon) [reply src/testing/taylor_green_vortex_energy_check.cpp:25]: it's a value generated by the code when I run the viscous TGV case, the test is simply to check that any future additions to the code do not modify the "expected" values.

**Comment by jbrillon** [comment]
Will mark this PR ready for review once the `_LONG` test passed, and after both have re-passed after rebasibg the branch

**Pull request closed without merging**